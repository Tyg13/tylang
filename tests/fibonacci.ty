fn printf(fmt: str, ...) -> i32 = extern;

fn fibonacci(n: i64) -> i64 = {
    if n <= 1 {
        return 1;
    }
    fibonacci(n - 1) + fibonacci(n - 2)
}

fn main(argc: i32, argv: *str) -> i32 = {
    printf("fib(2) == %d\n", fibonacci(2));
    printf("fib(5) == %d\n", fibonacci(5));
    0
}
