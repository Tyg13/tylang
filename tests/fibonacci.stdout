fib(2) == 2
fib(5) == 8
